boot/bootstate20.go

```diff
@@ -659,6 +718,62 @@ func (bs20 *bootState20Base) commit() error {
 	return nil
 }
 
+// selectAndCommitSnapInitramfsMount chooses which snap should be mounted
+// during the early boot sequence, i.e. the initramfs, and commits that
+// choice if it needs state updated.
+// Choosing to boot/mount the base snap needs to be committed to the
+// modeenv, but no state needs to be committed when choosing to mount a
+// kernel snap.
+func (bs20 *bootState20Base) selectAndCommitSnapInitramfsMount() (sn snap.PlaceInfo, err error) {
+	err = bs20.loadModeenv()
+	if err != nil {
+		return nil, err
+	}
+
+	// first do the generic choice of which snap to use
+	// the logic in that function is sufficient to pick the base snap entirely,
+	// so we don't ever need to look at the fallback snap, we just need to know
+	// whether the chosen snap is a try snap or not, if it is then we process
+	// the modeenv in the "try" -> "trying" case
+	first, _, currentlyTryingSnap, err := genericInitramfsSelectSnap(bs20, TryStatus, "base")
+	if err != nil {
+		return nil, err
+	}
+
+	modeenvChanged := false
+
+	// apply the update logic to the choices modeenv
+	switch bs20.modeenv.BaseStatus {
+	case TryStatus:
+		// if we were in try status and we have a fallback, then we are in a
+		// normal try state and we change status to TryingStatus now
+		// all other cleanup of state is left to user space snapd
+		if currentlyTryingSnap {
+			bs20.modeenv.BaseStatus = TryingStatus
+			modeenvChanged = true
+		}
+	case TryingStatus:
+		// we tried to boot a try base snap and failed, so we need to reset
+		// BaseStatus
+		bs20.modeenv.BaseStatus = DefaultStatus
+		modeenvChanged = true
```

Comment:
it's already cleared in the bootState20MarkSuccessful.commit method which will be eventually called during snapd startup from boot.MarkBootSuccessful, so we don't need to clear it here. We need to clear/reset BaseStatus here though because that is used to determine other logic. 

I think as a design point, we should have the initramfs mutate as little as possible, so it feels better to not clean this here and let snapd userspace cleanup. Additionally, this logic matches the uc16 case, where the bootloader does not reset snap_try_core, i.e. see https://github.com/snapcore/pc-amd64-gadget/blob/16/grub.cfg.